library/network/src/modules/NetworkInterfaces.rb

```diff
@@ -822,6 +824,16 @@ def CleanCacheRead
       Read()
     end
 
+    # Returns a hash with configuration for particular device
+    #
+    # Hash map is direct maping of sysconfig file into hash.
+    # Keys are sysconfig options (e.g. { 'IPADDR' => '1.1.1.1' }
+    #
+    # @return [Hash] device configuration or nil in case of error
+    def devmap(name)
+      Devices().fetch(GetType(name), {})[name]
```

Comment:
Wasn't this the cause of some bugs in the past? I mean, why use the GetType to obtain the ifcfg map instead of look based on ifcfg-name ?

Reply:
no the issue was caused by the fact that the cache was constructed using type detection based on ifcfg only. Then it could happen (and happened) that searching for device configuration failed bcs according to ifcfg it was recognized as (e.g.) vlan but in fact according to hwinfo (/sys) it was (e.g.) eth. This was fixed by PRs you refer bellow. Now is cache constructed according to same rules which ```GetType``` uses for type recognition (look into /sys first, ifcfg as a backup only). 

Anyway, this PR only moves code from y2-network (```GetDeviceMap```) into yast2 package (```devname```). I think that this code do not belong to y2-network bcs it looks into internal structure of NetworkInterface's cache. I use the bug reference only due to BS politics. The fix itself is in PRs referenced bellow.

Reply:
Yep, if now both rely on same type detection there should not be inconsistencies anymore, and that was the main reason why in `GetDeviceMap` the map was search just matching the ifcfg name and not relying in the ifcfg type anymore.

From performance POV it is better to use the type index, just hope it does not break anymore because of that...